transaction/std/src/transaction_builder.rs

```diff
@@ -505,7 +505,7 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
 /// * `fog_hint` - The encrypted fog hint to use
 /// * `memo_fn` - The memo function to use -- see TxOut::new_with_memo docu
 /// * `rng` -
-fn create_output_with_fog_hint<RNG: CryptoRng + RngCore>(
+pub(crate) fn create_output_with_fog_hint<RNG: CryptoRng + RngCore>(
```

Comment:
What's the motivation for making them public only within the crate again? I'm sure there's a great motivation I'm just not grokking the reasoning for it immediately.

Reply:
i did this because I needed this for the `test_utils` module, it uses it when making test data. Previously those functions were in `mod tests` in this file, and submodules of a module don't need `pub(crate)` to see things in the parent modules. But sibling modules can only see eachother's things if they are at least at `pub(crate)` visibility.